webrtc/RTCDataChannel-send.html

```diff
@@ -0,0 +1,380 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>RTCDataChannel.prototype.send</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="RTCPeerConnection-helper.js"></script>
+<script>
+  'use strict';
+
+  // Test is based on the following editor draft:
+  // https://w3c.github.io/webrtc-pc/archives/20170605/webrtc.html
+
+  // The following helper functions are called from RTCPeerConnection-helper.js:
+  //  createDataChannelPair
+  //  awaitMessage
+  //  blobToArrayBuffer
+  //  assertEqualsArrayBuffer
+
+  /*
+    6.2.  RTCDataChannel
+      interface RTCDataChannel : EventTarget {
+        ...
+        readonly  attribute RTCDataChannelState readyState;
+        readonly  attribute unsigned long       bufferedAmount;
+                  attribute EventHandler        onopen;
+                  attribute EventHandler        onmessage;
+                  attribute DOMString           binaryType;
+
+        void send(USVString data);
+        void send(Blob data);
+        void send(ArrayBuffer data);
+        void send(ArrayBufferView data);
+      };
+
+      bufferedAmount
+        The bufferedAmount attribute must return the number of bytes of application
+        data (UTF-8 text and binary data) that have been queued using send() but that,
+        as of the last time the event loop started executing a task, had not yet been
+        transmitted to the network. (This thus includes any text sent during the
+        execution of the current task, regardless of whether the user agent is able
+        to transmit text asynchronously with script execution.) This does not include
+        framing overhead incurred by the protocol, or buffering done by the operating
+        system or network hardware. If the channel is closed, this attribute's value
+        will only increase with each call to the send() method (the attribute does not
+        reset to zero once the channel closes).
+
+
+    [WebMessaging]
+    interface MessageEvent : Event {
+      readonly attribute any data;
+      ...
+    };
+   */
+
+  // Simple ASCII encoded string
+  const helloString = 'hello';
+  // ASCII encoded buffer representation of the string
+  const helloBuffer = Uint8Array.of(0x68, 0x65, 0x6c, 0x6c, 0x6f);
```

Comment:
Technically we could use the `TextEncoder` to encode the ASCII and unicode strings. But I think hard coded byte values is better here, since the test strings are short. It avoids additional dependency and demonstrates that nothing can go wrong of using the encoder incorrectly.